packages/@uppy/dashboard/src/index.js

```diff
@@ -923,7 +933,9 @@ module.exports = class Dashboard extends Plugin {
       this.uppy.use(ThumbnailGenerator, {
         id: `${this.id}:ThumbnailGenerator`,
         thumbnailWidth: this.opts.thumbnailWidth,
-        waitForThumbnailsBeforeUpload: this.opts.waitForThumbnailsBeforeUpload
+        waitForThumbnailsBeforeUpload: this.opts.waitForThumbnailsBeforeUpload,
+        // If we don't block on thumbnails, we can lazily generate them
+        lazy: !this.opts.waitForThumbnailsBeforeUpload
```

Comment:
Should we add a warning to the ThumbnailGenerator that if `{ waitForThumbnailsBeforeUpload: true, lazy: true }`, - then thumbnails will never get generated?

Otherwise I'd probably remove the .lazy option altogether. Cause:
`lazy: !this.opts.waitForThumbnailsBeforeUpload` is the most common use case,
`{ waitForThumbnailsBeforeUpload: true, lazy: true }` won't work, and
`{ waitForThumbnailsBeforeUpload: false, lazy: false }` - is valid, but I can't see the use case.